code/modules/cargo/centcom_podlauncher.dm

```diff
@@ -589,49 +684,75 @@
-					launchList |= typecache_filter_list_reverse(acceptableTurfs[launchCounter].contents, ignored_atoms) //Filter the specicic turf chosen from acceptableTurfs, and add it to the launchList
+				var/turf/T = acceptableTurfs[launchCounter]
+				for (var/atom/movable/O in T.contents) //Go through the acceptableTurfs list based on the launchCounter index
+					launchList |= typecache_filter_list_reverse(T.contents, ignored_atoms) //Filter the specicic turf chosen from acceptableTurfs, and add it to the launchList
+				if (iswallturf(T))
+					launchList += T
 			if(2) //If we are launching randomly
-				launchList |= typecache_filter_list_reverse(pick_n_take(acceptableTurfs).contents, ignored_atoms) //filter a random turf from the acceptableTurfs list and add it to the launchList
+				var/turf/T = pick_n_take(acceptableTurfs)
+				launchList |= typecache_filter_list_reverse(T.contents, ignored_atoms) //filter a random turf from the acceptableTurfs list and add it to the launchList
+				if (iswallturf(T))
+					launchList += T
 	updateSelector() //Call updateSelector(), which, if we are launching one at a time (launchChoice==2), will move to the next turf that will be launched
 	//UpdateSelector() is here (instead if the if(1) switch block) because it also moves the selector to nullspace (to hide it) if needed
 
 /datum/centcom_podlauncher/proc/launch(turf/A) //Game time started
 	if (isnull(A))
 		return
 	var/obj/structure/closet/supplypod/centcompod/toLaunch = DuplicateObject(temp_pod) //Duplicate the temp_pod (which we have been varediting or configuring with the UI) and store the result
-	if(dropoff_turf)
-		toLaunch.reverse_dropoff_turf = dropoff_turf
-	else
-		toLaunch.reverse_dropoff_turf = bay //Bay is currently a nonstatic expression, so it cant go into toLaunch using DuplicateObject
 	toLaunch.update_icon()//we update_icon() here so that the door doesnt "flicker on" right after it lands
 	var/shippingLane = GLOB.areas_by_type[/area/centcom/supplypod/supplypod_temp_holding]
 	toLaunch.forceMove(shippingLane)
 	if (launchClone) //We arent launching the actual items from the bay, rather we are creating clones and launching those
 		if(launchRandomItem)
-			var/atom/movable/O = pick_n_take(launchList)
-			DuplicateObject(O).forceMove(toLaunch) //Duplicate a single atom/movable from launchList and forceMove it into the supplypod
+			var/atom/O = pick_n_take(launchList)
+			if(!isnull(O))
+				if (iswallturf(O))
+					toLaunch.turfs_in_cargo += O.type
+				else
+					var/atom/movable/AM = O
+					DuplicateObject(AM).forceMove(toLaunch) //Duplicate a single atom/movable from launchList and forceMove it into the supplypod
 		else
-			for (var/atom/movable/O in launchList)
-				DuplicateObject(O).forceMove(toLaunch) //Duplicate each atom/movable in launchList and forceMove them into the supplypod
+			for (var/atom/O in launchList)
+				if (iswallturf(O))
+					toLaunch.turfs_in_cargo += O.type
+				else
+					var/atom/movable/AM = O
```

Comment:
Better var names.
Also, can there be things that are not `/atom` in `launchList`? If not, there's no need to type check.

Reply:
wouldn't I need to have `var/atom/O` in order to call `O.type`?